### ISSUE TITLE
Delay after scrolling over compositor borders

### ISSUE THREAD

**Issue by narfg** [servo/servo#5960]
After pushing the Page Up or Page Down keys there is a delay of about 10s before the page scrolls. Other key presses (e.g. typing in a text field) don't seem to show this behavior.
I see this issue in debug and release builds on Linux.

**Comment by jdm** [comment]
Does it matter what page you're looking at? This is presumably either the script task being very busy (and therefore not receiving the key event for some time), or the compositor being very busy (and therefore not receiving the key event reply for some time).

**Comment by narfg** [comment]
The issue does not seem to depend on the content or the complexity of the page. I can reproduce the issue on simple pages like tests/html/lipsum.html.
I investigated a little bit further and can give you some more details.
It's not an issue with scroll keys as it also happens with mouse scrolling (it's just harder to reproduce because of #5660). 
The delay happens exactly after scrolling over the compositor borders (verified with -Z show-compositor-borders).
I first noticed this bug after the rustc update two days ago. Before that, scrolling worked quite well. To confirm this, I checked out 7b87085c1880c60aa3be5b3ec4572a0d93fd5537 and I can't reproduce the issue there.
During the 10s delay there is no significant CPU usage, so I can't say if any part of servo is actually busy.
